lib/puppet/type.rb

```diff
@@ -1316,6 +1318,9 @@ def pathbuilder
       and the second run will log the edit made by Puppet.)"
 
     validate do |list|
+      if Puppet.settings[:strict] != :off
+        Puppet.deprecation_warning(_("You are attempting to use the `audit` metaparameter in %{resource} in %{manifest}. This parameter is deprecated and will be ignored in a future release.") % { resource: @resource.path, manifest: @resource.catalog.environment_instance.manifest })
```

Comment:
Is there a better way to say where this warning comes from? I was just using pry to poke around in the resource object, but there's a lot of stuff there...